app/routes/_layout+/bookings.new.tsx

```diff
@@ -154,17 +157,23 @@ export async function action({ context, request }: ActionFunctionArgs) {
       senderId: authSession.userId,
     });
 
-    const manageAssetsUrl = `/bookings/${
-      booking.id
-    }/add-assets?${new URLSearchParams({
-      // We force the as Date because we know that the booking.from and booking.to are set and exist at this point.
-      bookingFrom: (booking.from as Date).toISOString(),
-      bookingTo: (booking.to as Date).toISOString(),
-      hideUnavailable: "true",
-      unhideAssetsBookigIds: booking.id,
-    })}`;
-
-    return redirect(manageAssetsUrl);
+    const url = new URL(request.url);
+    const assetIds = url.searchParams.get("assetIds");
```

Comment:
That is not the correct way to do it in the action. We don't send search params to actions. You need to add it to the formData and parse the array of ids using zod. You can see how we do it in the bulk actions. For an example you can check `tags.bulk-actions.ts`

Also you should pass the `assetIds` to upsert booking. The way it is now the assets won't get added.

Reply:
Parsed assetsId array using zod and pass assetIds to upsert. Resolved!